usb/usb.py

```diff
@@ -0,0 +1,245 @@
+    if len(lines) > 2:
+        return lines[1].rstrip("\n")
+    else:
+        return ""
+
+def getUnlockedCryptOutput(path):
+    mountPoint = getMountPoint(path)
+    if mountPoint:
+        color = MOUNTED_COLOR
+        if isReadOnly(path):
+            spaceAvail = "ro"
+        else:
+            spaceAvail = getSpaceAvailable(path)
+        mountPoint = "<i>{}</i>:".format(pangoEscape(mountPoint))
+    else:
+        color = UNLOCKED_NOT_MOUNTED_COLOR
+        spaceAvail = ""
+    kernelName = getKernelName(path)
+    parentKernelName = getParentKernelName(path)
+    
+    block = "<span color='{}'>[{} {}:{}]</span>"
+    block = block.format(color, FA_UNLOCK, parentKernelName, kernelName)
+
+    label = pangoEscape(getFSLabel(path))
+    if label:
+        label = '"{}"'.format(label)
+
+    items = [block, label, mountPoint, spaceAvail]
+    return " ".join(filter(None, items))
+
+def getSwapOutput(path):
+    return ""
+
+def getUnencryptedPartitionOutput(path): 
+    mountPoint = getMountPoint(path)
+    if mountPoint:
+        color = MOUNTED_COLOR
+        if isReadOnly(path):
+            spaceAvail = "ro" 
+        else:
+            spaceAvail = getSpaceAvailable(path)
+        mountPoint = "<i>{}</i>:".format(pangoEscape(mountPoint))
+    else:
+        color = PLUGGED_COLOR
+        spaceAvail = ""
+    kernelName = getKernelName(path)
+
+    block = "<span color='{}'>[{}]</span>"
+    block = block.format(color, kernelName)
+
+    label = pangoEscape(getFSLabel(path))
+    if label:
+        label = '"{}"'.format(label)
+
+    items = [block, label, mountPoint, spaceAvail]
+    return " ".join(filter(None, items))
+
+def getDiskWithNoPartitionsOutput(path):
+    form = "<span color='{}'>[{}] {}</span>"
+    kernelName = getKernelName(path)
+    return form.format(PARTITIONLESS_COLOR, kernelName, PARTITIONLESS_TEXT)
+
+def getOutput(path):
+    if isSwapPartition(path):
+        return getSwapOutput(path)
+    t = getDeviceType(path)
+    if t == "part":
+        if isExtendedPartitionMarker(path):
+            return ""
+        elif isLUKSPartition(path):
+            return getLockedCryptOutput(path)
+        else:
+            return getUnencryptedPartitionOutput(path)
+    elif t == "disk":
+        return getDiskWithNoPartitionsOutput(path)
+    elif t == "crypt":
+        return getUnlockedCryptOutput(path)
+    elif t == "rom" :
+        return ""
+
+def makeAttributeMap(path):
+    attributeMap = {}
+    lines = check_output(
+            ['udevadm','info','--query=property','--name={}'.format(path)],
+            universal_newlines=True)
+    lines = lines.split("\n")
+    for line in lines:
+        if line:
+            key, val = line.split("=", maxsplit=1)
+            attributeMap[key] = val
+    return attributeMap
+
+def getAttributeMaps(paths):
+    return { path : makeAttributeMap(path) for path in paths}
+
+leaves = getLeafDevicePaths()
+leaves = list(filter(lambda path: not fastIgnore(path), leaves))
```

Comment:
May I suggest `leaves = [path for path in leaves if not fastIgnore(path)]` with removes both list and filter (and the lambda)?